### DISCUSSION TITLE
Broker.xml custom configuration in a single deployment

### DISCUSSION THREAD

**Discussion by JoseFSilva01** [arkmq-org/arkmq-org-broker-operator#525] [Q&A] [answered]
Hello,
My name is Jose Silva and i am a software engineer associate/ devops support and I currently have an artemis container in kubernetes with no operator. I was wondering if its possible to configure the broker xml file in a single kubernetes broker deployment of artemis with no operator?

**Comment by brusdev** [comment, marked as answer]
Hi @JoseFSilva01, the main purpose of this project is to provide an operator to deploy a cluster of brokers on Kubernetes. Your use case is not supported but you could execute `kubectl run` using the [activemq-artemis-broker-kubernetes-image](https://github.com/artemiscloud/activemq-artemis-broker-kubernetes-image) to deploy a a single broker with no operator, for further details see https://github.com/artemiscloud/activemq-artemis-broker-image/issues/22#issuecomment-1404009737

**Reply by JoseFSilva01** [reply]
Thank you i just wanted to make sure if maybe through a dockerfile maybe we could build it out some way but it gets overwritten everytime. so ill just stick to the operator method thank you!

**Reply by JoseFSilva01** [reply]
sorry to bother you again but what do i insert into the custom configuration file exactly?

**Reply by brusdev** [reply]
A custom `configure_custom_config.sh` file can be used to edit or overwrite the broker configuration files. i.e.
```
#!/bin/sh
set -e

INSTANCE_DIR=$1

cat <<EOT > $INSTANCE_DIR/etc/broker.xml
<?xml version='1.0'?>
<configuration xmlns="urn:activemq"
               xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
               xmlns:xi="http://www.w3.org/2001/XInclude"
               xsi:schemaLocation="urn:activemq /schema/artemis-configuration.xsd">
   <core xmlns="urn:activemq:core" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="urn:activemq:core ">
         <!-- my costom settings -->
   </core>
</configuration>
EOT
```

```
docker run -it --rm --name artemis -e AMQ_USER=admin -e AMQ_PASSWORD=admin -p 8161:8161 -v $(pwd)/configure_custom_config.sh:/opt/amq/bin/configure_custom_config.sh quay.io/artemiscloud/activemq-artemis-broker-kubernetes:latest
```
